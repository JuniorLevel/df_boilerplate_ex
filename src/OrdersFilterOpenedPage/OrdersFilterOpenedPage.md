```jsx { "file": "./OrdersFilterOpenedPage.jsx" }

```

```jsx noeditor
import Image1 from './ordersFilter-opened1.png';
import Image2 from './ordersFilter-opened2.png';
<div>
	<div>
		<img src={Image1} />
	</div>
	<div>
		<img src={Image2} />
	</div>
</div>;
```
