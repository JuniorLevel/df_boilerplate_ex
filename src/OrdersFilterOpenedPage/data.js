export const sidebarButtons = [
	{
		id: 1,
		title: 'Orders',
		onClick: () => {},
	},
	{
		id: 2,
		title: 'Compare',
		onClick: () => {},
	},
	{
		id: 3,
		title: 'Admin',
		onClick: () => {},
	},
];

export const filters = [
	{
		id: 1,
		cardItem: [{ id: 1, status: 'changed', text: 'lorem-1' }],
	},

	{
		id: 2,
		cardItem: [
			{ id: 1, status: 'updated', text: 'project-1' },
			{ id: 2, status: 'changed', text: 'project-2' },
			{ id: 3, status: 'updated', text: 'project-3' },
		],
	},
];
