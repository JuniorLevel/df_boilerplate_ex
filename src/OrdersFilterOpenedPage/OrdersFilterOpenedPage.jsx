// @flow

import React from 'react';
import { OrdersFilterOpened, Layout, OrdersFilterProvider } from '@/components';
import { filters, sidebarButtons } from './data';

export const OrdersFilterOpenedPage = (): React.Node => (
	<OrdersFilterProvider>
		<Layout buttons={sidebarButtons} title="Caption" avatar={{ title: 'AV' }}>
			<OrdersFilterOpened open dialog={false} filters={filters} />
		</Layout>
	</OrdersFilterProvider>
);
