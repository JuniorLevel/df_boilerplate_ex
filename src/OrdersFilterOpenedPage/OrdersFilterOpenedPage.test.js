import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { render } from '@testing-library/react';
import { OrdersFilterOpenedPage } from './OrdersFilterOpenedPage';
import '../../__mocks__/fileMock';

describe('OrdersFilterOpenedPage', () => {
	test('should render OrdersFilterOpenedPage component without crashing', () => {
		const { container } = render(<OrdersFilterOpenedPage />);
		expect(container).toBeInTheDocument();
	});
	test('matches the snapshot desktop', () => {
		const { container: desktop } = render(
			<ResponsiveContext.Provider value={{ width: 1920 }}>
				<OrdersFilterOpenedPage />
			</ResponsiveContext.Provider>
		);
		expect(desktop).toMatchSnapshot();
	});
	test('matches the snapshot laptop', () => {
		const { container: laptop } = render(
			<ResponsiveContext.Provider value={{ width: 1024 }}>
				<OrdersFilterOpenedPage />
			</ResponsiveContext.Provider>
		);
		expect(laptop).toMatchSnapshot();
	});
	test('matches the snapshot mobile', () => {
		const { container: mobile } = render(
			<ResponsiveContext.Provider value={{ width: 360 }}>
				<OrdersFilterOpenedPage />
			</ResponsiveContext.Provider>
		);
		expect(mobile).toMatchSnapshot();
	});
});
