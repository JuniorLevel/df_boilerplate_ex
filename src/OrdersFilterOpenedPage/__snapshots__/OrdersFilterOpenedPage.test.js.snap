// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersFilterOpenedPage matches the snapshot desktop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      />
    </div>
  </div>
</div>
`;

exports[`OrdersFilterOpenedPage matches the snapshot laptop 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      />
    </div>
  </div>
</div>
`;

exports[`OrdersFilterOpenedPage matches the snapshot mobile 1`] = `
<div>
  <div
    class="ant-layout Layoutstyles__StyledLayout-sc-httflb-0 dCkqfg css-1gwfwyx"
  >
    <div
      class="ant-layout css-1gwfwyx"
    >
      <main
        class="ant-layout-content Layoutstyles__StyledContent-sc-httflb-1 eUqUAz css-1gwfwyx"
      />
    </div>
  </div>
</div>
`;
